examples/fortran/util.f90

```diff
@@ -0,0 +1,244 @@
+        integer(kind=c_int64_t)         :: mb, nb, lda
+        integer(kind=c_int64_t)         :: i, j
+
+        do j = 0, slate_Matrix_nt_c32( A )-1
+            do i = 0, slate_Matrix_mt_c32( A )-1
+                if (slate_Matrix_tileIsLocal_c32( A, i, j )) then
+                    T = slate_Matrix_at_c32( A, i, j )
+                    mb = slate_Tile_mb_c32( T )
+                    nb = slate_Tile_nb_c32( T )
+                    lda = slate_Tile_stride_c32( T )
+                    T_data_c = slate_Tile_data_c32( T )
+                    call c_f_pointer( T_data_c, T_data, [lda*nb] )
+
+                    call random_matrix_c32( mb, nb, T_data, lda )
+                end if
+            end do
+        end do
+    end subroutine
+
+    subroutine random_matrix_type_c64( A )
+        use, intrinsic :: iso_fortran_env
+        use slate
+        implicit none
+
+        type(c_ptr), value              :: A
+        type(slate_Tile_c64)            :: T
+        type(c_ptr)                     :: T_data_c
+        complex(kind=c_double), pointer :: T_data(:)
+        integer(kind=c_int64_t)         :: mb, nb, lda
+        integer(kind=c_int64_t)         :: i, j
+
+        do j = 0, slate_Matrix_nt_c64( A )-1
+            do i = 0, slate_Matrix_mt_c64( A )-1
+                if (slate_Matrix_tileIsLocal_c64( A, i, j )) then
+                    T = slate_Matrix_at_c64( A, i, j )
+                    mb = slate_Tile_mb_c64( T )
+                    nb = slate_Tile_nb_c64( T )
+                    lda = slate_Tile_stride_c64( T )
+                    T_data_c = slate_Tile_data_c64( T )
+                    call c_f_pointer( T_data_c, T_data, [lda*nb] )
+
+                    call random_matrix_c64( mb, nb, T_data, lda )
+                end if
+            end do
+        end do
+    end subroutine
+
+    subroutine grid_size( mpi_size, p, q )
+        use, intrinsic :: iso_fortran_env
+        implicit none
+
+        integer(kind=c_int) :: mpi_size, p, q
+        real                :: mpi_size_real
+
+        mpi_size_real = mpi_size
+        p = floor( sqrt( mpi_size_real ) )
+        do while (p > 0)
+            q = mpi_size / p
+
+            if (p == q) then
+               return
+            end if
+        end do
+
+        ! no perfect grid found
+        print '(a, i1, a)', 'could not find a grid for ', mpi_size, 'ranks'
+        stop 1
+
+    end subroutine
+
+    subroutine parse_args( types )
+
+        logical           :: types(4)
+        character(len=64) :: arg
+
+        if (command_argument_count() == 1) then
+            types( 1 ) = .true.
+            types( 2 ) = .true.
+            types( 3 ) = .true.
+            types( 4 ) = .true.
+        else
+            types( 1 ) = .true.
+            types( 2 ) = .true.
+            types( 3 ) = .true.
+            types( 4 ) = .true.
+        endif
+
+        do i = 1, command_argument_count()
+            call getarg( 0, arg )
```

Comment:
Should be `getarg( i, arg )`, instead of `0`, right?
But `get_command_argument( i, arg )` would be preferred for Fortran 2003.
Fixed.